src/app/model-poller.js

```diff
@@ -21,46 +21,72 @@ import {
 
 import { userIsControllerAdmin } from "./utils";
 
-export default async function connectAndListModels(reduxStore, bakery) {
+export default async function connectAndListModels(
+  reduxStore,
+  bakery,
+  additionalControllers
+) {
   try {
     const storeState = reduxStore.getState();
-    const credentials = getUserPass(storeState);
     const { identityProviderAvailable, isJuju } = getConfig(storeState);
     const wsControllerURL = getWSControllerURL(storeState);
-    const { conn, error, juju, intervalId } = await loginWithBakery(
+    const credentials = getUserPass(wsControllerURL, storeState);
+    const defaultControllerData = [
       wsControllerURL,
       credentials,
       bakery,
-      identityProviderAvailable
-    );
-    if (error) {
-      reduxStore.dispatch(storeLoginError(error));
-      return;
+      identityProviderAvailable,
+    ];
+    let controllerList = [defaultControllerData];
+    if (additionalControllers) {
+      controllerList = controllerList.concat(additionalControllers);
     }
-    if (process.env.NODE_ENV === "production") {
-      Sentry.setTag("jujuVersion", conn?.info?.serverVersion);
-    }
-    reduxStore.dispatch(updateControllerConnection(conn));
-    reduxStore.dispatch(updateJujuAPIInstance(juju));
-    reduxStore.dispatch(updatePingerIntervalId(intervalId));
-    if (userIsControllerAdmin(conn)) {
-      fetchControllerList(conn, reduxStore);
-      if (!isJuju) {
-        // This call will be a noop if the user isn't an administrator
-        // on the JIMM controller we're connected to.
-        disableControllerUUIDMasking(conn);
+    controllerList.forEach(async (controllerData) => {
+      const { conn, error, juju, intervalId } = await loginWithBakery(
+        ...controllerData
+      );
+
+      if (error) {
+        reduxStore.dispatch(storeLoginError(error));
+        return;
       }
-    }
-    do {
-      await reduxStore.dispatch(fetchModelList());
-      await fetchAllModelStatuses(conn, reduxStore);
-      // Wait 30s then start again.
-      await new Promise((resolve) => {
-        setTimeout(() => {
-          resolve(true);
-        }, 30000);
-      });
-    } while (isLoggedIn(reduxStore.getState()));
+
+      if (process.env.NODE_ENV === "production") {
```

Comment:
We do this check before initing Sentry at root level.. is it also needed here? If so, what can we do at the conditional at root level to ensure calling `Sentry` just gracefully fails if not on production?

Reply:
We need to connect to the controller first to be able to get its version so it has to be done here.

We could probably create a wrapper around sentry which would have this check built in. But that would be for a follow-up, feel free to file an issue for that.